src/awkward/contents/recordarray.py

```diff
@@ -1115,6 +1115,13 @@ def _to_backend_array(self, allow_missing, backend):
         contents = [x._to_backend_array(allow_missing, backend) for x in self._contents]
         if any(len(x.shape) != 1 for x in contents):
             raise ValueError(f"cannot convert {self} into np.ndarray")
+
+        if not backend.nplike.supports_structured_dtypes:
+            raise TypeError(
+                f"backend {backend.name!r} does not support structured "
+                f"dtypes required for converting {type(self).__name__} "
+                f"into a backend array"
+            )
```

Comment:
You noticed this: okay, so maybe I didn't need to tell you.

Reply:
This is a cleanup more than a feature; NumPy _does_ support both DLPack and structured dtypes. However, NumPy will report that the conversion fails, e.g.
```
>>> import awkward as ak
>>> array = ak.Array([{'x': 1}]).to_numpy()
>>> ptr = array.__dlpack__()
Traceback (most recent call last):
  File "...", line 5, in <module>
    ptr = array.__dlpack__()
          ^^^^^^^^^^^^^^^^^^
BufferError: DLPack only supports signed/unsigned integers, float and complex dtypes.
[('x', '<i8')]
```